flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-maxcompute/src/main/java/org/apache/flink/cdc/connectors/maxcompute/sink/MaxComputeEventWriter.java

```diff
@@ -0,0 +1,190 @@
+        this.context = context;
+        this.options = options;
+        this.writeOptions = writeOptions;
+
+        this.writerMap = new HashMap<>();
+        this.schemaCache = new HashMap<>();
+    }
+
+    @Override
+    public void write(Event element, Context context) throws IOException {
+        if (element instanceof DataChangeEvent) {
+            DataChangeEvent dataChangeEvent = (DataChangeEvent) element;
+            String sessionId = dataChangeEvent.meta().get(Constant.TUNNEL_SESSION_ID);
+            String partitionName = dataChangeEvent.meta().get(Constant.MAXCOMPUTE_PARTITION_NAME);
+            if (!writerMap.containsKey(sessionId)) {
+                LOG.info(
+                        "Sink writer {} start to create session {}.",
+                        this.context.getSubtaskId(),
+                        sessionId);
+                SessionIdentifier sessionIdentifier =
+                        SessionIdentifier.of(
+                                options.getProject(),
+                                MaxComputeUtils.getSchema(options, dataChangeEvent.tableId()),
+                                dataChangeEvent.tableId().getTableName(),
+                                partitionName,
+                                sessionId);
+                writerMap.put(
+                        sessionId,
+                        MaxComputeWriter.batchWriter(options, writeOptions, sessionIdentifier));
+            }
+            MaxComputeWriter writer = writerMap.get(sessionId);
+            ArrayRecord record = writer.newElement();
+
+            if (dataChangeEvent.op() != OperationType.DELETE) {
+                TypeConvertUtils.toMaxComputeRecord(
+                        schemaCache.get(dataChangeEvent.tableId()),
+                        dataChangeEvent.after(),
+                        record);
+                writer.write(record);
+            } else {
+                TypeConvertUtils.toMaxComputeRecord(
+                        schemaCache.get(dataChangeEvent.tableId()),
+                        dataChangeEvent.before(),
+                        record);
+                writer.delete(record);
+            }
+        } else if (element instanceof CreateTableEvent) {
+            CreateTableEvent createTableEvent = (CreateTableEvent) element;
+            schemaCache.put(createTableEvent.tableId(), createTableEvent.getSchema());
+        } else if (element instanceof SchemaChangeEvent) {
+            SchemaChangeEvent schemaChangeEvent = (SchemaChangeEvent) element;
+            TableId tableId = schemaChangeEvent.tableId();
+            Schema newSchema =
+                    SchemaUtils.applySchemaChangeEvent(schemaCache.get(tableId), schemaChangeEvent);
+            schemaCache.put(tableId, newSchema);
+        }
+    }
+
+    @Override
+    public void flush(boolean endOfInput) throws IOException, InterruptedException {
+        SessionManageOperator operator = SessionManageOperator.instance;
```

Comment:
We can add Preconditions.checkNotNull() check here to guide users avoid setting configuration like 'pipeline.operator-chaining'='false'

Reply:
ok